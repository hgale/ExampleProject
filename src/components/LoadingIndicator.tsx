import * as React from 'react';
import { ActivityIndicator, StyleSheet, Text, View, ViewStyle } from 'react-native';
import { colors } from 'src/utils/styles';

interface Props {
  style?: ViewStyle;
}

const LoadingIndicator = (props: Props) => (
  <View style={ styles.container }>
    <ActivityIndicator
      style={ [ props.style, styles.indicator] }
      size='large'
      color={ colors.LOADING }
    />
    <Text style={ styles.loadingText }>Loading...</Text>
  </View> 
);

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.BACKGROUND,
  },
  loadingText: {
    color: colors.WHITE,
  },
  indicator: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 8,
  },
});

export default LoadingIndicator;
