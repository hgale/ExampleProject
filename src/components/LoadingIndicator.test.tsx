import * as React from 'react';
import 'react-native';
import * as renderer from 'react-test-renderer';

import LoadingIndicator from 'src/components/LoadingIndicator';

it('renders correctly', () => {
  const component = renderer.create(<LoadingIndicator />);
  expect(component).toMatchSnapshot();
});
