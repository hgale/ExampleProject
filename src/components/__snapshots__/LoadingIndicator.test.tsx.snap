// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(55, 57, 62)",
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="1B9ECC"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        undefined,
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "padding": 8,
        },
      ]
    }
  />
  <Text
    style={
      Object {
        "color": "rgb(255, 255, 255)",
      }
    }
  >
    Loading...
  </Text>
</View>
`;
